File: spring-mvc/src/main/java/org/example/JsonConfig.java
package org.example;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;

@Configuration
public class JsonConfig {
    /**
     * 나는 자비빈이 싫어요
     *
     * @see "https://youtube.com/watch?v=dQw4w9WgXcQ"
     */
    @Bean
    public MappingJackson2HttpMessageConverter mappingJackson2HttpMessageConverter() {
        MappingJackson2HttpMessageConverter converter = new MappingJackson2HttpMessageConverter();
        converter.setObjectMapper(new ObjectMapper()
                .configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false));
        return converter;
    }
}

File: spring-mvc/src/main/kotlin/JsonConfig.kt
package org.example

import kotlinx.serialization.json.Json
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.converter.HttpMessageConverter
import org.springframework.http.converter.json.KotlinSerializationJsonHttpMessageConverter

@Configuration
open class JsonConfig {
	@Bean
	open fun json(): Json = Json {
		prettyPrint = true
		isLenient = true
		ignoreUnknownKeys = true
	}
	
	@Bean
	open fun kotlinSerializationJsonHttpMessageConverter(json: Json): HttpMessageConverter<*> =
		KotlinSerializationJsonHttpMessageConverter(json)
	
}